data-generation/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+    // Interval is defined on the top of the file
+    let distr_90p = Uniform::<i64>::try_from((expected_amnt * BOTTOM / 100)..(expected_amnt * UPPER / 100)).unwrap();
+    let sampled_expected = distr_90p.sample(&mut rng);
+
+    // I will store rows as columns in form of vectors
+    let ids: Vec<i64> = vec![id; sampled_expected.as_usize()];
+    let mut card_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut trx_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut ch_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut amnt: Vec<f64> = Vec::with_capacity(sampled_expected.as_usize());
+    let t_minus: Vec<i64> = vec![d_minus; sampled_expected.as_usize()];
+
+
+    for _i in 0..sampled_expected {
+        // Choose DC with probability 75%
+        let card_type = if rng.gen_bool(0.75) {
+            CARD_TYPES.0
+        } else {
+            CARD_TYPES.1
+        };
+
+        // Choose random transaction type
+        let trx_type = TRANSACTION_TYPES
+            .get(distr_cat.sample(&mut rng).as_usize())
+            .unwrap();
+
+        // Choose mobile channel with probability 25%
+        let channel_type = if rng.gen_bool(0.25) {
+            CHANNELS.0
+        } else {
+            CHANNELS.1
+        };
+
+        // Sample from uniform distribution for an amount
+        let trx_amnt = distr_trx.sample(&mut rng);
+
+        card_t.push(card_type);
+        trx_t.push(trx_type);
+        ch_t.push(channel_type);
+        amnt.push(trx_amnt);
+    }
+
+    (ids, card_t, trx_t, ch_t, amnt, t_minus)
+}
+
+/// Generate batch of random seeded transactions for the given customers list considering expected amount of transactions
+#[pyfunction]
+fn generate_data_batch(
+    ids: PyArrowType<ArrayData>,
+    trx_per_day: PyArrowType<ArrayData>,
+    days_in_batch: i64,
+    offset: i64,
+    global_seed: u64,
+    partition_name: &str,
+) -> PyArrowType<RecordBatch> {
+    let ids_arr = make_array(ids.0);
+    let expected_arr = make_array(trx_per_day.0);
+    let dim = ids_arr.len() * days_in_batch.as_usize();
+
+    let trx_amnt_distr = Uniform::<f64>::try_from(MIN..MAX).unwrap();
+    let trx_cat_distr = Uniform::<i64>::try_from(0..12).unwrap();
+
+    // I will store generated batches as vector of vectors
+    let mut ids_out_col: Vec<Vec<i64>> = Vec::with_capacity(dim);
+    let mut card_type_out_col: Vec<Vec<&str>> = Vec::with_capacity(dim);
+    let mut trx_type_out_col: Vec<Vec<&str>> = Vec::with_capacity(dim);
+    let mut channel_out_col: Vec<Vec<&str>> = Vec::with_capacity(dim);
+    let mut trx_amnt_out_col: Vec<Vec<f64>> = Vec::with_capacity(dim);
+    let mut t_minus_out_col: Vec<Vec<i64>> = Vec::with_capacity(dim);
+
+    // This generator is used for generating of u64 seeds for mini-batch generation
+    let mut global_rnd = ChaCha8Rng::seed_from_u64(global_seed);
+
+    let iter = ArrayIter::new(
```

Comment:
I think it can be done shorter, `Int64Array::from` with something, I would recommend to ask ChatGPT.

Reply:
Thank you! I found sampling from slice in `rand`, it looks better and it seems to me that it works faster!